runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -10570,6 +10570,310 @@ inlineCompareAndSwapNative(
+   dependencies->addPostCondition(loopLimitReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(limitReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(maskReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(indexReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(xmmChunkReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(chunkReg, TR::RealRegister::NoReg, cg);
+   dependencies->stopAddingConditions();
+
+   // Labels
+   TR::LabelSymbol *begLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *endLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *loopLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *residualLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *threeOrMoreBytesLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *fiveOrMoreBytesLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *nineOrMoreBytesLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *residualTestLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *returnNoNegativesLabel = NULL;
+   TR::LabelSymbol *returnHasNegativesLabel = NULL;
+   TR::LabelSymbol *returnBooleanLabel = NULL;
+   if (recognizedMethod == TR::java_lang_StringCoding_hasNegatives)
+      {
+      returnBooleanLabel = generateLabelSymbol(cg);
+      }
+   else
+      {
+      returnNoNegativesLabel = generateLabelSymbol(cg);
+      returnHasNegativesLabel = generateLabelSymbol(cg);
+      }
+   begLabel->setStartInternalControlFlow();
+   endLabel->setEndInternalControlFlow();
+
+   // A key part of the main loop of this algorithm is the pmovmskb instruction,
+   // which extracts the sign bits from a source register and collects them into a destination register
+   // We need to specify the encoding of this instruction so that the code generator knows we want the 16 byte version
+   TR::InstOpCode pmovmskb = TR::InstOpCode::PMOVMSKB4RegReg;
+   OMR::X86::Encoding pmovmskbEncoding = pmovmskb.getSIMDEncoding(&cg->comp()->target().cpu, TR::VectorLength128);
+
+
+   generateLabelInstruction(TR::InstOpCode::label, node, begLabel, cg);
+
+   // index = offset
+   generateRegRegInstruction(TR::InstOpCode::MOV4RegReg, node, indexReg, offsetReg, cg);
+
+   // limit = offset + length
+   generateRegMemInstruction(TR::InstOpCode::LEA4RegMem, node, limitReg, generateX86MemoryReference(offsetReg, lengthReg, 0, 0, cg), cg);
+
+   // loopLimit = (length & -16) + offset
+   generateRegRegInstruction(TR::InstOpCode::MOV4RegReg, node, loopLimitReg, lengthReg, cg);
+   generateRegImmInstruction(TR::InstOpCode::AND4RegImm4, node, loopLimitReg, -16, cg);
+   generateRegRegInstruction(TR::InstOpCode::ADD4RegReg, node, loopLimitReg, offsetReg, cg);
+
+   // If the 16 byte encoding of the pmovmskb instruction is not supported on this architecture,
+   // Prepare an 8 byte sign bit mask so we can run an alternate version of the algorithm
+   if (pmovmskbEncoding == OMR::X86::Encoding::Bad)
```

Comment:
May I ask you to introduce a `bool` variable to give this condition a name &mdash; `useEightByteChunks` or `useVectorRegister`, say &mdash; and then use that variable in this test and the remaining tests below?

It might also be useful to introduce an environment variable that can be used to force the version that does not use `pmovmskb` to be generated.